scripts/fgi

```diff
@@ -0,0 +1,259 @@
+#!/bin/bash
+# Copyright (c) The Diem Core Contributors
+# SPDX-License-Identifier: Apache-2.0
+set -e
+set -o pipefail
+
+TAG=""
+PR=""
+WORKSPACE=""
+ENV=""
+LOCAL_BUILD=""
+EXIT_CODE=0
+# Default timeout is 45 mins
+TIMEOUT_SECS=2700
+
+FORGE_K8S_CLUSTERS=(rustietest)
+K8S_CONTEXT_PATTERN='arn:aws:eks:us-west-2:853397791086:cluster/libra-CLUSTERNAME'
+HELM_CHARTS_URL_PATTERN="s3://diem-testnet-CLUSTERNAME-helm/charts"
+GRAFANA_URL_PATTERN="http://mon.CLUSTERNAME.aws.hlw3truzy4ls.com"
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+# Colorize Output
+RESTORE=$(echo -en '\001\033[0m\002')
+BLUE=$(echo -en '\001\033[01;34m\002')
+
+join_args() {
+  retval_join_args=""
+  for var in "$@"
+  do
+    retval_join_args="${retval_join_args}, \"${var}\""
+  done
+}
+
+join_env_vars() {
+  retval_join_env_vars=""
+  for var in "$@"
+  do
+    IFS='=' read -ra env_var <<< "$var"
+    retval_join_env_vars="{\"name\":\"${env_var[0]}\", \"value\":\"${env_var[1]}\"}, ${retval_join_env_vars}"
+  done
+}
+
+# https://stackoverflow.com/a/5533586
+# This function shuffles the elements of an array named "array" in-place using the Knuth-Fisher-Yates shuffle algorithm
+shuffle() {
+   local i tmp size max rand
+
+   # $RANDOM % (i+1) is biased because of the limited range of $RANDOM
+   # Compensate by using a range which is a multiple of the array size.
+   size=${#array[*]}
+   max=$(( 32768 / size * size ))
+
+   for ((i=size-1; i>0; i--)); do
+      while (( (rand=RANDOM) >= max )); do :; done
+      rand=$(( rand % (i+1) ))
+      tmp=${array[i]} array[i]=${array[rand]} array[rand]=$tmp
+   done
+}
+
+# init the kube context for each available cluster
+kube_init_context () {
+  aws eks --region us-west-2 describe-cluster --name "libra-${FORGE_K8S_CLUSTERS[0]}" &>/dev/null || (echo "Failed to access EKS, try awsmfa?"; exit 1)
+  for cluster in "${FORGE_K8S_CLUSTERS[@]}"; do
+    aws eks --region us-west-2 update-kubeconfig --name "libra-${cluster}"
+  done
+}
+
+# randomly select a cluster that is free
+kube_select_cluster () {
```

Comment:
S3 based locking was for an ECS testnet compat test that was eventually scrapped. This current method of selecting clusters isn't perfect but has worked for cluster-test for some time :p It's also simpler since we don't have to grant as many permissions